packages/govuk-frontend/src/govuk/govuk-frontend-component.mjs

```diff
@@ -8,27 +8,43 @@ import { InitError, SupportError } from './errors/index.mjs'
  *
  * @internal
  * @virtual
+ * @template [T=HTMLElement]
  */
 export class GOVUKFrontendComponent {
+  /**
+   * @protected
+   * @type {T}
+   */
+  $root
+
   /**
    * Constructs a new component, validating that GOV.UK Frontend is supported
    *
    * @internal
-   * @param {Element | null} [$root] - HTML element to use for component
+   * @param {T | null} [$root] - HTML element to use for component
    */
   constructor($root) {
     const childConstructor = /** @type {ChildClassConstructor} */ (
       this.constructor
     )
 
+    if (!($root instanceof HTMLElement)) {
+      throw new ElementError({
+        component: childConstructor,
+        identifier: 'Root element (`$root`)'
+      })
+    } else {
+      this.$root = $root
+    }
+
     childConstructor.checkSupport()
 
     this.checkInitialised($root)
```

Comment:
**question** I think we have a choice to make as to which feature runs first between checking initialisation and storing the root:
1. If we store the root first, `checkInitialised` can then benefit from having the root stored and not receive an argument
2. If we check initialisation first, we need to keep the argument, but we'll only store the root if we're actually initialising the component.

2. feels the more natural order to me, what are your thoughts?